Salehi2025/data.R

```diff
@@ -17,237 +17,127 @@ library(glmnet)
-
-
-# This section prepares the data for alpha diversity analysis and 
-# computes Shannon diversity index from relative abundance profiles
-
-# Filter metadata to include only samples with GDP and antibiotic usage information
+# Filter samples with complete GDP and antibiotic usage data
+# and ensure continent is stored as a factor variable
 adult_metadata <- adult_metadata %>%
-  mutate(
-    continent = factor(geo_loc_name_country_continent_calc)
-  ) %>%
+  mutate(continent = factor(geo_loc_name_country_continent_calc)) %>%
   filter(!is.na(Usage), !is.na(GDP_per_head))
 
-# Keep only samples with non-missing GDP and antibiotic usage in the TSE object
-non_na_conditions <- !is.na(colData(TSE_filtered)$GDP_per_head) &
-  !is.na(colData(TSE_filtered)$Usage)
-TSE_adult_filtered <- TSE_filtered[, non_na_conditions]
-
-# Update the sample metadata in the filtered TSE object
-colData(TSE_adult_filtered) <- DataFrame(adult_metadata)
-
-# Extract relative abundance data
-assay_data_clean <- assay(TSE_adult_filtered, "relabundance")
-
-# Remove samples that contain only zero abundances
-non_empty_samples <- colSums(assay_data_clean > 0, na.rm = TRUE) > 0
-assay_data_clean <- assay_data_clean[, non_empty_samples]
-
-# Subset metadata to match samples with non-zero data
-adult_metadata <- colData(TSE_adult_filtered)[non_empty_samples, ]
-
-# Calculate Shannon diversity for each sample (based on species richness and evenness)
-adult_metadata$shannon_diversity <- diversity(t(assay_data_clean), index = "shannon")
-
+# Subset TSE to retain only samples present in filtered metadata
+tse_class_top5 <- tse_class_top5[, colnames(tse_class_top5) %in% adult_metadata$acc]
+colData(tse_class_top5) <- DataFrame(adult_metadata)
 
-# This section creates dummy variables (one-hot encoding) for categorical 
-# predictors to be used in the GLM and Bayesian regression models
+# Remove samples with zero total relative abundance across all features
+tse_class_top5 <- tse_class_top5[, colSums(assay(tse_class_top5, "relabundance")) > 0]
 
-# Copy metadata to a temporary dataframe
-temp_df <- adult_metadata
-
-# Set the correct factor level order for age categories
-temp_df$age_category <- factor(temp_df$age_category,
-                               levels = c("Middle-Aged Adult", "Infant",
-                                          "Toddler", "Children", "Teenager",
-                                          "Young Adult", "Older Adult", "Oldest Adult"))
-
-# Create binary variable: High income (1) vs. other (0)
-temp_df$income_group_HIC <- ifelse(temp_df$World_Bank_Income_Group == "High income", 1, 0)
-
-# Create binary variable for antibiotic usage: high (1) vs. low (0)
-temp_df$Usage_high <- ifelse(temp_df$Usage < 10, 0, 1)
+# Calculate Shannon diversity index directly using mia's built-in support
+tse_class_top5 <- addAlpha(
+  tse_class_top5,
+  assay.type = "relabundance",
+  index = "shannon",
+  name = "shannon_diversity"
+)

```

Comment:
This can be problematic: alpha diversity should be calculated from the original count assay with as high taxonomic resolution as possible (and not from top-aggregated data).

I suggest to calculate Shannon diversity for the data (-> to add in colData) already in the data preparation script from the original full "counts" assay and then you can use that for all analyses where Shannon diversity is used.